e2e/specs/add-appointments.spec.ts

```diff
@@ -0,0 +1,114 @@
+import { expect } from '@playwright/test';
+import { generateRandomPatient, deletePatient, type Patient } from '../commands';
+import { test } from '../core';
+import { AppointmentsPage } from '../pages';
+
+let patient: Patient;
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+});
+
+test('Add appointment for a patient, edit the added appointment and cancel it', async ({ page, api }) => {
```

Comment:
I would suggest we re-title the test to

```
Add, edit and cancel an appointment
```

Reply:
I would suggest this be broken up into three tests, as it's three features:

1. Add an appointment
2. Edit an existing appointment
3. Cancel an existing appointment

Reply:
> I would suggest this be broken up into three tests, as it's three features:
> 
> 1. Add an appointment
> 2. Edit an existing appointment
> 3. Cancel an existing appointment

@ibacher we are following this [doc](https://docs.google.com/document/d/1fvq0DH3NJSCxThe1jYlv3TqbvgW7OO3QswYQcocbHiM/edit#heading=h.o4hk4pnqimq7)  for the tests. So for adding tests for appointment section, is it better to do it as a whole test for appointment section?

Reply:
> I would suggest this be broken up into three tests, as it's three features:

@jayasanka-sack what do you think concerning Ian's idea?

Reply:
The reason I suggested to use a single flow is to minimize API usage for data population, but after seeing how lengthy the test going to be, I also think breaking into three tests would make it much simpler.

Reply:
We've got just a handful of tests that contain in single flow. Let's leave them as is for the time being and beef them up later when we can. Our priority is to expand our coverage. But, if you happen to have some extra time, feel free to tweak them too.

Reply:
There's definitely a balance of concerns, but if the main concern is how much data is created during, it should be possible, for example, to create a single patient and run a suite of tests using that single patient (i.e., there's no reason here that the patient needs to be created in a `beforeEach()` rather than a `beforeAll()`, especially since we're fundamentally only working on data attached to the patient record rather than the patient record itself).

My concern here is related to this point in that doc: "We want business-level users to easily be able to see a report that reassures them about what functionality is covered". Specifically, if the "Add appointment for a patient, edit the added appointment and cancel it" test _fails_, then, as a business user, I'm stuck regarding that _whole feature_  as broken, which is much more concerning than saying "We can currently add appointments, but something broke the cancellation flow", for example.

I also agree with @jayasanka-sack's point that the priority should be improving test coverage, not going back and revising everything to match some "gold standard" for how tests are written.

Reply:
Created a [ticket ](https://openmrs.atlassian.net/browse/O3-3064)to monitor the proposed E2E test approach.